Jenkinsfile

```diff
@@ -0,0 +1,24 @@
+pipeline {
```

Comment:
I would prefer the stages to be triggered by make command like is used to be. 
preparing virtualenv shouldn't be a separate stage. I would see it more like a dependency for other stages.
I suggest to configure separate jobs and statuses for independent tasks like style, unit tests, functional tests. This way in github statues there would be better visibility which task failed.